### PR TITLE
Support Enterprise GitHub

### LINKED ISSUE #4: Enterprise GitHub URLs are not supported
Trying to use `default-branch` with Enterprise GitHub URLs causes error.

```javascript
defaultBranch('https://enterprise.github.corp/org/repo').then(branch => {
    console.log(branch);
}).catch(err => {
    console.error('Error');
    // => Error
});
```
The reason is the malformed URL being constructed:
```
https://github.com/https://enterprise.github.corp/org/repo/branches
```

### PR DESCRIPTION
**Pull request by micellius**
* add Enterprise GitHub support
* update tests and docs
* bump version to 1.0.9
* fix #4

### CODE REVIEW

**Comment by knutkirkhorn** [comment]
Thank you for this! 😄 I am really sorry for the very very late response.
Will take a look at this now :thumbsup:

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "default-branch",
-  "version": "1.0.8",
+  "version": "1.0.9",
```
Comment (knutkirkhorn) [inline package.json]: Can you revert the version upgrade? I would like to put it in a separate commit and probably bump the version further.

**Review by knutkirkhorn — CHANGES_REQUESTED** [review]
Looks really good, thanks! 😄

File: package.json (hunk 1 above)
Comment (micellius) [reply package.json]: Sure, no problem. Already done :)

Comment (knutkirkhorn) [reply package.json]: Thanks! 😄
